packages/tools/src/eventListeners/annotations/contourSegmentation/contourSegmentationCompleted.ts

```diff
@@ -0,0 +1,216 @@
+    );
+
+    const targetAABB = math.polyline.getAABB(targetPolyline);
+    const polylinesIntersect =
+      math.aabb.intersectAABB(sourceAABB, targetAABB) &&
+      math.polyline.intersectPolyline(sourcePolyline, targetPolyline);
+
+    if (polylinesIntersect) {
+      return { targetAnnotation, targetPolyline };
+    }
+  }
+}
+
+function processContours(
+  viewport: Types.IViewport,
+  sourceAnnotation: ContourSegmentationAnnotation,
+  sourcePolyline: Types.Point2[],
+  targetAnnotation: ContourSegmentationAnnotation,
+  targetPolyline: Types.Point2[]
+) {
+  const sourceStartPoint = sourcePolyline[0];
+  const mergePolylines = math.polyline.containsPoint(
+    targetPolyline,
+    sourceStartPoint
+  );
+
+  const newPolylines = [];
+
+  if (mergePolylines) {
+    const mergedPolyline = math.polyline.mergePolylines(
+      targetPolyline,
+      sourcePolyline
+    );
+
+    newPolylines.push(mergedPolyline);
+  } else {
+    const subtractedPolylines = math.polyline.subtractPolylines(
+      targetPolyline,
+      sourcePolyline
+    );
+
+    subtractedPolylines.forEach((newPolyline) =>
+      newPolylines.push(newPolyline)
+    );
+  }
+
+  removeAnnotation(sourceAnnotation.annotationUID);
+  removeAnnotation(targetAnnotation.annotationUID);
+
+  const { element } = viewport;
+  const enabledElement = getEnabledElement(element);
+  const { renderingEngine } = enabledElement;
+  const { metadata, data } = targetAnnotation;
+  const { handles, segmentation } = data;
+  const { textBox } = handles;
+
+  newPolylines.forEach((newPolyline) => {
+    const polyline = newPolyline.map((p) => viewport.canvasToWorld(p));
+    const startPoint = polyline[0];
+    const endPoint = polyline[polyline.length - 1];
+    const newAnnotation = {
+      metadata: {
+        ...metadata,
+        toolName: DEFAULT_CONTOUR_SEG_TOOLNAME,
+      },
+      data: {
+        cachedStats: {},
+        handles: {
+          points: [startPoint, endPoint],
+          textBox: textBox ? { ...textBox } : undefined,
+        },
+        contour: {
+          polyline,
+          closed: true,
+        },
+        segmentation: {
+          ...segmentation,
+        },
+      },
+      annotationUID: csUtils.uuidv4() as string,
+      highlighted: true,
+      invalidated: true,
+      isLocked: false,
+      isVisible: undefined,
+    };
+
+    addAnnotation(newAnnotation, element);
+
+    // Updating a Spline contours, for example, should also update freehand contours
+    const updatedTtoolNames = new Set([
+      DEFAULT_CONTOUR_SEG_TOOLNAME,
+      targetAnnotation.metadata.toolName,
+    ]);
+
+    for (const toolName of updatedTtoolNames.values()) {
+      const viewportIdsToRender = getViewportIdsWithToolToRender(
+        element,
+        toolName
+      );
+      triggerAnnotationRenderForViewportIds(
+        renderingEngine,
+        viewportIdsToRender
+      );
+    }
+  });
+}
+
+export default function contourSegmentationCompletedListener(
```

Comment:
I expect the default function to be the first thing in a file when I open it instead of noisy functions and then the default. Also docs please